keras/src/models/functional.py

```diff
@@ -494,16 +494,28 @@ def process_layer(layer_data):
             # (e.g. a model such as A(B(A(B(x)))))
             add_unprocessed_node(layer, node_data)
 
+    # Extract config used to instantiate Functional model from the config. The
+    # remaining config will be passed as keyword arguments to the Model
+    # constructor.
+    functional_config = {}
+    for key in ["layers", "input_layers", "output_layers"]:
```

Comment:
You can simply this block:

- single iteration over set of keys
- use `pop(key, None)`

Reply:
I have applied your suggestion, I like the idea to simplify it. However, it does not keep the previous behavior. Now instead of throwing a `KeyError` if `layers`, `input_layers` or `output_layers` is not present, it will silently set its value to `None`. It doesn't seem to be an issue, but I just wanted to mention it.

Reply:
You're right, it would lead to a strange error message instead of a clear one if the config is malformed. Let's avoid that.

Reply:
I reverted the commit then. I am still waiting for the CLA to be handled. I'll let you know when that's done.